bundles/org.openhab.binding.blink/src/main/java/org/openhab/binding/blink/internal/servlet/ThumbnailServlet.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.blink.internal.servlet;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.blink.internal.handler.CameraHandler;
+import org.osgi.service.http.HttpService;
+import org.osgi.service.http.NamespaceException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ThumbnailServlet} class provides the servlet for outputting a camera's current thumbnail.
+ *
+ * @author Matthias Oesterheld - Initial contribution
+ */
+@NonNullByDefault
+public class ThumbnailServlet extends HttpServlet {
+
+    public static final long serialVersionUID = 666L;
+
+    private final Logger logger = LoggerFactory.getLogger(ThumbnailServlet.class);
+    private final HttpService httpService;
+    private final CameraHandler cameraHandler;
+    private final String servletUrl;
+
+    public ThumbnailServlet(HttpService httpService, CameraHandler cameraHandler) {
+        this.httpService = httpService;
+        this.cameraHandler = cameraHandler;
+
+        try {
+            servletUrl = "/blink/thumbnail/"
+                    + URLEncoder.encode(cameraHandler.getThing().getUID().getId(), StandardCharsets.UTF_8);
+            httpService.registerServlet(servletUrl, this, null, httpService.createDefaultHttpContext());
```

Comment:
Make the servlet name unique to prevent issues when cameras are used.

See the [community topic](https://community.openhab.org/t/blink-binding/132328/47)

```suggestion
            servletUrl = "/blink/thumbnail/"
                    + URLEncoder.encode(cameraHandler.getThing().getUID().getId(), StandardCharsets.UTF_8);

            Hashtable<Object, Object> initParams = new Hashtable<>();
            initParams.put("servlet-name", servletUrl);

            httpService.registerServlet(servletUrl, this, initParams, httpService.createDefaultHttpContext());
```